lib/jsonapi/paginator.ex

```diff
@@ -0,0 +1,20 @@
+defmodule JSONAPI.Paginator do
```

Comment:
I think I understand what's going on here. Would I be correct in saying that this module should be implemented by the consumer of this lib to generate these links?

Reply:
A module conforming to the `Paginator` behavior can be either configured as the global pagination logic in `config :jsonapi, ;paginator, MODULE`, and/or passed to a view via `use JSONAPI.View, paginator: MODULE`. Its implementation can be done by the consumer of the library and/or by the library itself. I think shipping implementations for common pagination schemes (page/offset/cursor) with the library could be valueable, but I'd love to hear from you on this topic. This is also something that could be discussed separately from this PR.

Reply:
> This is also something that could be discussed separately from this PR.

Totally agreed. I'd rather see this come in to the lib first, have some people play with it and iron out basic bugs.

Reply:
Ok, I'm going to move the page based implementation from the code to the docs as an example and we can figure out if including implementations with the library is feasible at a later time.